File: 02 tests/ex1_sum_of_squares.py
#!/usr/bin/env python3

""" ex1_sum_of_squares.py

Sum of squares

:author:	wolf
:created:	2025.05.22
"""

import time

iterations = 3
timing_results = []

# define the test function
def python_sum_squares(n):
    return sum(i * i for i in range(1, n + 1))


for i in range(iterations):
    start = time.time()

    # call the test function
    python_sum_squares(10**9)

    end = time.time()

    timing_results.append(end - start)


print(f"Elapsed time: {sum(timing_results)/iterations:.4f} seconds")

File: 02 tests/ex1_sum_of_squares.js
// run with: deno run ex1_sum_of_squares.js

/* ex1_sum_of_squares.js

Sum of squares

:author:  wolf
:created: 2025.05.22
*/

const iterations = 3;
const timing_results = [];

// define the test function
// --- function to be timed-- -

function jsSumSquares(n) {
    let sum = 0;
    for (let i = 1; i <= n; i++) {
        sum += i * i;
    }
    return sum;
}


for (let i = 0; i < iterations; i++) {
    const start = performance.now();

    // call the test function
    jsSumSquares(10 ** 9);

    const end = performance.now();

    timing_results.push(end - start);
}


console.log("Timing results: ",
    timing_results.reduce((a, b) => a + b) / iterations / 1000,
    "seconds");